docs/sources/setup/kubernetes-helm.md

```diff
@@ -0,0 +1,142 @@
+* It tries to instrument all the applications in your cluster.
+* It only provides application-level metrics. [Network-level metrics]({{< relref "../network" >}}) are excluded by default.
+* Configures Beyla to decorate the metrics with Kubernetes metadata labels (`k8s.namespace.name`, `k8s.pod.name`, etc.).
+
+## Configuring Beyla
+
+You might want to override the default configuration of Beyla. For example, to export the metrics and/or spans
+as OpenTelemetry instead of Prometheus, or to restrict the number of services to instrument.
+
+You can override the default [Beyla configuration options]({{< relref "../configure" >}}) with your own values.
+
+For example, create a `helm-beyla.yml` file with a custom configuration:
+
+```yaml
+config:
+  data:
+    # Contents of the actual Beyla configuration file
+    discovery:
+      services:
+        - k8s_namespace: demo
+        - k8s_namespace: blog
+    routes:
+      unmatched: heuristic
+```
+
+The `config.data` section contains a Beyla configuration file whose options are documented in the
+[Beyla configuration options document]({{< relref "../configure/options.md" >}}).
+
+Then pass the overridden configuration to the `helm` command with the `-f` flag. For example:
+
+```
+helm install beyla grafana/beyla -f helm-beyla.yml
+```
+
+or, if the Beyla chart was previously deployed:
+
+```
+helm upgrade beyla grafana/beyla -f helm-beyla.yml
+```
+
+## Configuring Beyla metadata
+
+If Beyla exports the data using the Prometheus exporter, you might need to override the Beyla Pod
+annotations to let it be discoverable by your Prometheus scraper. You can add the following
+section to the example `helm-beyla.yml` file:
+
+```
+podAnnotations:
+  prometheus.io/scrape: 'true'
+  prometheus.io/path: '/metrics'
+  prometheus.io/port: '9090'
+```
+
+Analogously, the Helm chart allows overriding names, labels, and annotations for
+multiple resources that are involved in the deployment of Beyla, such as service
+accounts, cluster roles, security contexts, etc. The 
+[Beyla Helm chart documentation](https://github.com/grafana/beyla/blob/main/charts/beyla/README.md)
+describes the diverse configuration options.
+
+## Providing secrets to the Helm configuration
+
+If you are submitting directly the metrics and traces to Grafana Cloud via the
+OpenTelemetry Endpoint, you need to provide the credentials via the 
+`OTEL_EXPORTER_OTLP_HEADERS` environment variable.
+
+The recommended way is to store such value in a Kubernetes Secret and then
+specify the environment variable referring to it from the Helm configuration.
+
+For example, deploy the following secret: 
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: grafana-secret
+type: Opaque
+stringData:
+  otlp-headers: "Authorization=Basic ...."
+```
+
+Then refer to it from the `helm-config.yml` file via the `envValueFrom` section:
+
+```yaml
+env:
+  OTEL_EXPORTER_OTLP_ENDPOINT: "https://otlp-gateway-prod-eu-west-0.grafana.net/otlp"
```

Comment:
I think it might be safer if we made the OTEL_EXPORTER_OTLP_ENDPOINT a not working URL, since copy paste will work for EU users and won't for others. Maybe something like this:

```
env:
  OTEL_EXPORTER_OTLP_ENDPOINT: "<...your Grafana Cloud OTLP endpoint URL...> e.g. https://otlp-gateway-prod-eu-west-0.grafana.net/otlp"
```

What do you think?

Reply:
Has all the sense. Thanks for the suggestion!